utils/notifications.js

```diff
@@ -401,20 +401,31 @@ async function getParticipantsAndSendNotifications({ notificationSpec, cutoffTim
 
       try {
         await sgMail.send(emailBatch);
+        await saveNotificationBatch(emailRecordArray);
```

Comment:
would Promise.all() be a benefit here or 419 & 421?

Reply:
Since we only save email records after every thing is fine sending out emails, Promise.all() doesn't fit well here. Similar at line 419 and 421.